torch/quantization/fx/_equalize.py

```diff
@@ -220,3 +222,215 @@ def node_supports_equalization(node: Node, modules) -> bool:
     elif node.op == 'call_function':
         return node.target == nn.functional.linear
     return False
+
+def get_weight_eq_obs(node: Node, model: GraphModule, modules: Dict[str, nn.Module]):
+    """ Gets the following weight equalization observer. There should always
+    exsist a weight equalization observer after an input equalization observer.
+
+    Returns the node containing the weight equalization observer, and the weight
+    equalization observer if it has been newly created
+    """
+
+    # Find the next node that is either a nn.Linear or a functional layer
+    while (node.op not in ('output', 'call_function') and not
+           (node.op == 'call_module' and isinstance(node.target, str) and
+            isinstance(modules[node.target], nn.Linear))):
+        node = node.next
```

Comment:
how about this:

```
assert len(node.users.items()) == 1
first_user_node, _ = first(node.users.items())
# check that first_user_node is a linear
# get the weight

```

Reply:
I'm not sure how this could be extended to other models, but any suggestions would be appreciated <3

Currently the `node` (which contains the input equalization observer) is 2 "steps" away from the linear layer: `x -> InpEqObs -> InpQuantObs -> nn.Linear`
so we would need to do this procedure twice, maybe something like: 
```
next_node, _ = first(node.users.items())
# Check if next_node is a quantization observer
next_next_node, _ = first(next_node.users.items())
# Check if next_next_node is linear
```

However if the nn.Linear layers are connected, then the node containing the input equalization observer is only 1 "step" away from the linear layer: `inear1 -> OutQuantObs -> InpEqObs -> linear2`.
So the above code would work in getting the `linear2` node.

If the layers are F.linear, then instead of finding the layer, we just need to find the weight equalization observer (I think the code should be updated in the next PR), which might be hard to find in this format.
```
                                                         weight
                                                              |
                                                     WeightEqObs
                                                              |
                                                   WeightQuantObs
                                                              |
x -> InpEqObs -> InpQuantObs -> linear -> OutQuantObs
```

Reply:
> Currently the node (which contains the input equalization observer) is 2 "steps" away from the linear layer: x -> InpEqObs -> InpQuantObs -> nn.Linear
so we would need to do this procedure twice, maybe something like:

right, I was mentioning that it should be `x -> outputQuantObs -> inpEqObs -> nn.Linear`.  There is not a concept of "input quantization observer", and the equalization observer should go after the quantization observer.

Reply:
> However if the nn.Linear layers are connected, then the node containing the input equalization observer is only 1 "step" away from the linear layer: linear1 -> OutQuantObs -> InpEqObs -> linear2.

correct, can we make it work like this in all cases?

Reply:
Yup, with the new changes to `prepare`, we can now do this -- I have updated the code below 😄